src/triangle.rs

```diff
@@ -141,11 +120,17 @@ impl Triangle {
     #[doc(alias = "circumcircle")]
     #[inline]
     pub fn circumscribed_circle(&self) -> Circle {
-        let ab = self.a.distance(self.b);
-        let bc = self.b.distance(self.c);
-        let ac = self.a.distance(self.c);
+        let b = self.b - self.a;
+        let c = self.c - self.a;
+        let b_len2 = b.hypot2();
+        let c_len2 = c.hypot2();
+        let d_recip = 1. / (2. * b.cross(c));
 
-        Circle::new(self.circumcenter(), (ab * bc * ac) / (4.0 * self.area()))
+        let x = (c.y * b_len2 - b.y * c_len2) * d_recip;
+        let y = (b.x * c_len2 - c.x * b_len2) * d_recip;
+        let r = b_len2.sqrt() * c_len2.sqrt() * (c - b).hypot() * d_recip;
```

Comment:
If I'm reading this right, then a further possible optimization is to to use a single sqrt around the product of the distances squared of the sides.

Reply:
Well spotted! Nice.